src/FileAdder/FileAdder.php

```diff
@@ -331,4 +324,30 @@ protected function sanitizeFileName(string $fileName) : string
     {
         return str_replace(['#', '/', '\\'], '-', $fileName);
     }
+
+    /**
+     * Attach media to the model.
+     *
+     * @param $media
+     */
+    protected function attachMedia($media)
+    {
+        if (! $this->subject->exists) {
+            $this->subject->unsavedMedias[] = $media;
+
+            $class = get_class($this->subject);
+
+            $class::created(function ($model) {
+                if (! empty($model->unsavedMedias)) {
+                    foreach ($model->unsavedMedias as $unsavedMedia) {
+                        $model->media()->save($unsavedMedia);
+                        $this->filesystem->add($this->pathToFile, $unsavedMedia, $this->fileName);
+                    }
+                }
+            });
+        } else {
+            $this->subject->media()->save($media);
```

Comment:
These two lines appear twice in this function. Let's extract them to their own function. No sure on a good name here.